ttnn/cpp/ttnn/operations/pool/generic/device/pool_multi_core_program_factory.cpp

```diff
@@ -274,9 +310,11 @@ MaxPool2D::MultiCore::cached_program_t max_pool_2d_multi_core_sharded_with_halo_
     /**
      * Reader Kernel: input rows -> input cb
      */
-    float one = 1.;
-    uint32_t bf16_one_u32 = *reinterpret_cast<uint32_t*>(&one);
-    uint32_t in_nbytes_c_log2 = (uint32_t)std::log2((float)in_nbytes_c);
+    const float scalar = get_reduce_scalar(pool_type, kernel_size_hw);
+    const uint32_t bf16_scalar_u32 = *reinterpret_cast<const uint32_t*>(&scalar);
+    const float init_value = get_reduce_init_value(pool_type);
+    const uint32_t bf16_init_value_u32 = *reinterpret_cast<const uint32_t*>(&init_value);
```

Comment:
This does not seem right. Float -inf would be 32 bit value, reinterpreting that to cast to uint32_t would not make it -inf for bfp16. we can use converter, if not available then hard code them based on data type.

Reply:
I rewrote the conversions in `get_bf16_pool_scalar` and `get_bf16_pool_init_value` above, which uses `bfloat16(value).to_packed()` to convert float32 to bf16 and pack them into u16. The u16 is then unsigned-extended to u32 so they can be pass as kernel arguments. In kernel we truncate the u32 back to u16 to get the bf16 value